src/libraries/System.Private.CoreLib/src/System/IO/Directory.cs

```diff
@@ -315,5 +315,46 @@ public static string[] GetLogicalDrives()
         {
             return FileSystem.GetLogicalDrives();
         }
+
+        /// <summary>
+        /// Creates a directory symbolic link identified by <paramref name="path"/> that points to <paramref name="pathToTarget"/>.
+        /// </summary>
+        /// <param name="path">The absolute path where the symbolic link should be created.</param>
+        /// <param name="pathToTarget">The target directory of the symbolic link.</param>
+        /// <returns>A <see cref="DirectoryInfo"/> instance that wraps the newly created directory symbolic link.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="path"/> or <paramref name="pathToTarget"/> is <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentException"><paramref name="path"/> or <paramref name="pathToTarget"/> is empty.
+        /// -or-
+        /// <paramref name="path"/> is not an absolute path.
+        /// -or-
+        /// <paramref name="path"/> or <paramref name="pathToTarget"/> contains invalid path characters.</exception>
+        /// <exception cref="IOException">A file or directory already exists in the location of <paramref name="path"/>.
+        /// -or-
+        /// An I/O error occurred.</exception>
+        public static FileSystemInfo CreateSymbolicLink(string path, string pathToTarget)
+        {
+            FileSystem.VerifyValidPath(path, nameof(path));
+            FileSystem.VerifyValidPath(pathToTarget, nameof(pathToTarget));
+            FileSystem.CreateSymbolicLink(path, pathToTarget, isDirectory: true);
+            return new DirectoryInfo(path);
```

Comment:
`FileSystem.VerifyValidPath(path, nameof(path));` - the checks is in DirectoryInfo too (Path.GetFullPath(path) is called!) and there is expensive check `path.Contains('\0')` that we duplicate.
The same issue in `public static string GetFullPath(string path, string basePath)`.

I'd suggest refactor Path.GetFullPath so that have internal method without these checks. Then we could use internal DirectoryInfo() too:
```c#
return new DirectoryInfo(path, Path.InternalGetFullPath(path));
```
It seems there are other places where we could use the pattern.

Reply:
Good catch.

Reply:
I will fix the duplicated checks _here_ by calling `Path.GetFullPath` here (up-front) and then use the .ctors with `isNormalized: true` to bypass the `Path.GetFullPath` call performed there. 
@iSazonov can you please log an issue in case you know of any other places where we duplicate these checks?

Reply:
> @iSazonov can you please log an issue in case you know of any other places where we duplicate these checks?

Done #54993